internal/eval/eval.go

```diff
@@ -871,7 +871,9 @@ func evaluateAndSetExpiry(subCommands []string, newExpiry uint64, key string,
 			" GT or LT options at the same time are not compatible")
 	}
 
-	store.SetUnixTimeExpiry(obj, int64(newExpiry))
+	if shouldSetExpiry {
```

Comment:
Can we please add some unit/integration tests for this scenario?
Also can you confirm the successful run of the tcl test behavior. The steps to run the test cases are mentioned in the `README` of the [dice-tests repository](https://github.com/AshwinKul28/dice-tests).

Reply:
Yeah, I am on it. Will update the PR in some time. Thanks!